contrib/platform/src/com/sun/jna/platform/unix/LibCAPI.java

```diff
@@ -43,6 +43,33 @@
 
         private static final long serialVersionUID = 1L;
 
+        public static class ByReference extends com.sun.jna.ptr.ByReference {
+            public ByReference() {
+                this(0);
+            }
+
+            public ByReference(long value) {
+                this(new size_t(value));
+            }
+
+            public ByReference(size_t value) {
+                super(Native.SIZE_T_SIZE);
+                setValue(value);
+            }
+
+            public void setValue(size_t value) {
```

Comment:
I would add setters and getters from java long. A java long can cover the whole range.